src/subscription.js

```diff
@@ -0,0 +1,278 @@
+            event = args[3],
+            _in = args[4], 
+            targetLeague = args[5], 
+            _for = args[6];
+
+        // Ensure the basic command format is valid
+        if (!_.isEqual(["tell", "when", "in", "for"], [tell, when, _in, _for])) {
+            return formatHelpResponse(config).then(function(message) {
+                return "I didn't understand you.\n" + message;
+            });
+        }
+
+        // Ensure they chose a valid league.
+        var l = league.getLeague(targetLeague, config);
+        if (_.isUndefined(l)) {
+            return formatInvalidLeagueResponse(config);
+        }
+        // TODO: this is hacky, but required at the moment. :(
+        //       otherwise message.player.isModerator won't work.
+        //       this is usually set by the requiresLeague middleware,
+        //       but we are in a DM, not a league specific channel
+        message.league = l;
+
+        // TODO: Allow captains some amount of flexibility of who they can subscribe
+        if (!_.isEqual(listener, "me") && !message.player.isModerator()) {
+            return formatCanOnlyListenForYouResponse(config, listener);
+        }
+
+        // TODO: Allow more than just 'me' as a listener. For example:
+        //       - captains can subscribe a private channel of less than 10 people 
+        //         so long as they are a part of that channel themselves.
+        //       - moderators should be able to do the same.
+        if (!_.isEqual(listener, "me")) {
+            return formatInvalidListenerResponse(config, listener);
+        }
+
+        if (_.isEqual(listener, "me")) {
+            listener = requester.name;
+        }
+
+        // Ensure the event is valid
+        if (!_.includes(events, event)) {
+            return formatInvalidEventResponse(config, event);
+        }
+
+        // Ensure the source is a valid user within slack
+        // TODO: Allow teams as a source, not just users.
+        var source = player.getSlackUserFromNameOrID(slack.users, sourceName);
+        if (_.isUndefined(source)) {
+            return formatInvalidSourceResponse(config, sourceName);
+        }
+        return db.lock().then(function(unlock) {
+            return db.Subscription.findOrCreate({
+                where: {
+                    requester: requester.name.toLowerCase(),
+                    source: source.name.toLowerCase(),
+                    event: event.toLowerCase(),
+                    league: l.options.name.toLowerCase(),
+                    target: listener.toLowerCase()
+                }
+			}).then(function(subscription) {
+                unlock.resolve();
+                return "Great! I will tell {target} when {event} for {source} in {league}".format({
+                    source: source.name,
+                    event: event,
+                    league: l.options.name,
+                    target: listener
+                });
+            });
+        });
+    });
+}
```

Comment:
Looks like the basics are in place. Extension for teams and targets should be straightforward.